File: golang/cbnu_2021_2_online_contest/4/4.go
package main

import (
	"fmt"
)

func main() {
	var N int
	fmt.Scanln(&N)
	help := make([]string, N)
	var midterm, final, homework float64
	students := make([]int, N)
	score := make([]float64, N)
	var i int
	rank := make([]float64, N)
	for i = 0; i < N; i++ {
		fmt.Scanln(midterm, final, homework)
		score[i] = (0.35 * midterm) + (0.45 * final) + (0.2 * homework)
		students[i] = i + 1
	}
	for j := 0; j < N; j++ {
		for k := 0; k < N; k++ {
			if score[j]-score[k] < 0 {
				i += 1
			}
		}
		rank[i] = score[j]
		i = 0
	}
	temp := (len(students) / 10)
	temp = float64(temp)
	for i := 0; i < N; i++ {
		if rank[i] > 0 && rank[i] <= temp {
			help[i] = "A+"
		}
	}

}

File: golang/cbnu_2021_2_online_contest/4/4.cpp
#include <iostream>
#include <bits/stdc++.h>
using namespace std;
int main() {
	int T;
	scanf("%d", &T);
	int a[100];
	int N, K;
	int midterm, finalterm, homework;
	char grades[10][3] = {"A+","A0","A-","B+","B0","B-","C+","C0","C-","D0"};
	for(int i=1; i<=T; i++) {
		scanf("%d %d", &N, &K);
		for(int j=0; j<N; j++) {
			scanf("%d %d %d", &midterm, &finalterm, &homework);
			a[j] = midterm*35 + finalterm*45 + homework*20;
		}
		int score = a[K-1];
		int rank;
		sort(a, a+N);
		for(int j=0; j<N; j++) {
			if(a[j] == score) {
				rank = N-1-j;
				break;
			}
		}
		printf("#%d %s\n", i, grades[10*rank/N]);
	}
}